congame-doc/introduction.scrbl

```diff
@@ -1,37 +1,530 @@
+}
+
+Click the @onscreen{Start Survey} link/button to proceed to the next step in the survey:
+
+@browser{
+  @bold{@larger{Survey}}
+
+  What is your first name? @mock-textbox[]
+                       
+  What is your age in years?  @mock-textbox[]
+
+  @kbd{Submit}
+}
+
+Enter some values in the textboxes and click “Submit”.
+
+@margin-note{For fun, try entering a non-numeric value in the “age” text box. What happens when you
+click @onscreen{Submit}?}
+
+This will bring you to the final step in the study:
+
+@browser{
+  @bold{@larger{Good Job, Joel}}
+ 
+  Thank you for participating in our survey despite being 11 years old.
+}
+
+That’s it! You can see that the values you entered in the previous step have carried over and are
+being used in the text displayed in this step.
+
+@margin-note{When you upload and run the study on the Congame server, values collected and stored in
+variables created with @racket[defvar] and similar forms get recorded in the server’s database and
+comprise the results of the study.}
+
+You’ve now observed a few more basic concepts first-hand:
+
+@itemlist[
+
+@item{You use @racket[defvar] to define the discrete pieces of information you want to collect from
+study participants.}
+
+@item{You can create multiple @tech{steps} for your study through repeated use of @racket[defstep].}
+
+@item{Within steps, you can insert input controls like @racket[input-text] inside a @racket[form] to
+collect info from participants, and save that info into your @racket[defvar] variables.}
+
+@item{You use the @racket[defstudy] form to define your study as a whole. Within that form, you use
+@racketid[-->] to tell Congame how the steps connect to each other in sequences.}
+
+]
+
+It’s possible to create much more advanced studies than those we have built so far: studies with
+conditional branching, a variety of form controls, timers, randomizing assignments, etc. But what
+you have seen so far gives you the basic framework.
+
+@;===============================================
+
+@section{Uploading and running your study}
+
+When you have finished designing your study, it's time to go live and start collecting responses
+from participants. We’re going to do that now with the @filepath{age-survey.rkt} study we just
+created.
+
+In order to do this, you need access to a Congame server. For the steps that follow, we'll assume:
+
+@itemlist[#:style 'ordered
+
+@item{You have a link/address to a running Congame server.}
+
+@item{You already have a researcher or an administrator account on that server.}
+
+]
+
+@margin-note{See @seclink["congame-setup"] if you need to know how to prepare a server for use with
+studies.} 
+
+
+@; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+@subsection{Preparing the study: from @racketmodname[conscript/local] to @racketmodname[conscript]}
+
+@bold{Important:} before uploading the study to a Congame server, we need to change it to use 
+@code{#lang conscript} instead of @code{#lang conscript/local}.
```

Comment:
It is no longer necessary to remove the "/local", you should test that, but we allow both "conscript" and "conscript/local" and strip the "/local".

Reply:
In that case, is there ever a reason why I as a study author would want to use `conscript` rather than `conscript/local`?

Reply:
Currently, yes, since "conscript/local" is the one that is needed to run the correct #lang offline, which has a `preview` function. When we upload it, we literally strip the string "/local" and thereby trigger the correct #lang conscript to be run on the server.

However, yesterday we started implementing a dockerized offline solution that should replace conscript/local, because keeping `conscript/local' in synch with `conscript` is takes around 25-30% of development and has caused half the bugs in the last half year. Moving to docker and providing a thin command line utility to automate common workflows such as updating a local study will mean that we only have to maintain one version.

In short, I expect us to take all `conscript/local` stuff out in the next month, so don't invest time into it.